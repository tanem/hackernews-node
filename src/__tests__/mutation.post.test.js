const path = require('path')
const bcrypt = require('bcryptjs')
const chance = require('chance').Chance(path.basename(__filename))
const jwt = require('jsonwebtoken')
const { APP_SECRET } = require('../utils')
const request = require('supertest')

let httpServerUrl
let prisma
let token

beforeAll(async () => {
  httpServerUrl = global.__HTTP_SERVER_URL__
  prisma = global.__PRISMA__
})

const name = chance.first()
const email = chance.email()
const password = chance.word()

beforeEach(async () => {
  const { id: userId } = await prisma.user.create({
    data: {
      name,
      email,
      password: await bcrypt.hash(password, 10),
    },
  })
  token = jwt.sign({ userId }, APP_SECRET)
})

test('successful', async () => {
  const { body, status } = await request(httpServerUrl)
    .post('/')
    .set('Authorization', `Bearer ${token}`)
    .send({
      query: `
      mutation {
        post(
          url: "${chance.url()}"
          description: "${chance.sentence({ words: 5 })}"
        ) {
          id
          description
          url
          postedBy {
            id
            name
          }
          votes {
            id
            user {
              id
              name
            }
          }
        }
      }
    `,
    })

  expect(status).toBe(200)
  expect(body).toMatchSnapshot()
})

test('returns error if not authenticated', async () => {
  const { body, status } = await request(httpServerUrl)
    .post('/')
    .send({
      query: `
      mutation {
        post(
          url: "${chance.url()}"
          description: "${chance.sentence({ words: 5 })}"
        ) {
          id
          description
          url
          postedBy {
            id
            name
          }
          votes {
            id
            user {
              id
              name
            }
          }
        }
      }
    `,
    })

  expect(status).toBe(200)
  expect(body).toMatchSnapshot()
})
