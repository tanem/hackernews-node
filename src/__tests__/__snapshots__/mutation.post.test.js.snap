// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns error if not authenticated 1`] = `
Object {
  "data": null,
  "errors": Array [
    Object {
      "locations": Array [
        Object {
          "column": 9,
          "line": 3,
        },
      ],
      "message": "Not authenticated",
      "path": Array [
        "post",
      ],
    },
  ],
}
`;

exports[`successful 1`] = `
Object {
  "data": Object {
    "post": Object {
      "description": "Lub op so rotizjod loh.",
      "id": "1",
      "postedBy": Object {
        "id": "1",
        "name": "Lois",
      },
      "url": "http://jur.om/va",
      "votes": Array [],
    },
  },
}
`;
